tidy3d/components/data/monitor_data.py

```diff
@@ -575,6 +578,150 @@ class FluxTimeData(MonitorData):
+    >>> values = (1+1j) * np.random.random((len(orders_x), len(orders_y), len(pol), len(f)))
+    >>> field = DiffractionDataArray(values, coords=coords)
+    >>> monitor = DiffractionMonitor(
+    ...     center=(1,2,3), size=(np.inf,np.inf,0), freqs=f, name='diffraction',
+    ...     orders_x=orders_x, orders_y=orders_y
+    ...     )
+    >>> data = DiffractionData(
+    ...     monitor=monitor, E=field, H=field, ux=wave_numbers, uy=wave_numbers
+    ... )
+    """
+
+    monitor: DiffractionMonitor
+
+    ux: ArrayLike[float, 2] = pd.Field(
+        ...,
+        title="Normalized kx",
+        description="Local x component of wave vectors for each diffraction order and frequency, "
+        "relative to ``local_origin`` and oriented with respect to ``u_axis``, "
+        "normalized by (2*pi/lambda) where lambda is the wavelength "
+        "associated with the background medium.",
+    )
+
+    uy: ArrayLike[float, 2] = pd.Field(
+        ...,
+        title="Normalized ky",
+        description="Local y component of wave vectors for each diffraction order and frequency, "
+        "relative to ``local_origin`` and oriented with respect to ``u_axis``, "
+        "normalized by (2*pi/lambda) where lambda is the wavelength "
+        "associated with the background medium.",
+    )
+
+    E: DiffractionDataArray = pd.Field(
+        ...,
+        title="E",
+        description="Complex components of the electric far field for each polarization "
+        "tangential to ``monitor.normal_axis``, in the local coordinate system whose "
+        " z-axis is ``monitor.normal_axis``.",
+    )
+    H: DiffractionDataArray = pd.Field(
+        ...,
+        title="H",
+        description="Complex components of the magnetic far field for each polarization "
+        "tangential to ``monitor.normal_axis``, in the local coordinate system whose "
+        "z-axis is ``monitor.normal_axis``.",
+    )
+
+    @property
+    def field_components(self) -> Dict[str, DataArray]:
+        """Maps the field components to thier associated data."""
+        # pylint:disable=no-member
+        return {field: getattr(self, field) for field in ["E", "H"]}
+
+    def normalize(self, source_spectrum_fn: Callable[[float], complex]) -> DiffractionData:
+        """Return copy of self after normalization is applied using source spectrum function."""
+        fields_norm = {}
+        for field_name, field_data in self.field_components.items():
+            src_amps = source_spectrum_fn(field_data.f)
+            fields_norm[field_name] = field_data / src_amps
+
+        return self.copy(update=fields_norm)
+
+    def _make_coords(self):
+        """Helper to make the coordinates dictionary."""
+        coords = {}
+        coords["orders_x"] = np.atleast_1d(self.E.orders_x.values)
+        coords["orders_y"] = np.atleast_1d(self.E.orders_y.values)
+        coords["polarization"] = np.array(["s", "p"])
```

Comment:
since the coord can be lists and numpy arrays of strings can be a bit annoying to work with, might be better to just make this a list `coords['polarization'] = ['s', 'p']` instead.